common/src/main/java/com/wynntils/utils/rendering/Matrix4f.java

```diff
@@ -82,19 +84,42 @@ public boolean isInteger() {
+        return isInteger(matrix4f2.m00 / matrix4f2.m03)
+                && isInteger(matrix4f2.m10 / matrix4f2.m13)
+                && isInteger(matrix4f2.m20 / matrix4f2.m23)
+                && isInteger(matrix4f2.m01 / matrix4f2.m03)
+                && isInteger(matrix4f2.m11 / matrix4f2.m13)
+                && isInteger(matrix4f2.m21 / matrix4f2.m23)
+                && isInteger(matrix4f2.m02 / matrix4f2.m03)
+                && isInteger(matrix4f2.m12 / matrix4f2.m13)
+                && isInteger(matrix4f2.m22 / matrix4f2.m23);
     }
 
     private static boolean isInteger(float f) {
-        return (double)Math.abs(f - (float)Math.round(f)) <= 1.0E-5D;
+        return (double) Math.abs(f - (float) Math.round(f)) <= 1.0E-5D;
     }
 
     public boolean equals(Object object) {
         if (this == object) {
             return true;
         } else if (object != null && this.getClass() == object.getClass()) {
-            Matrix4f matrix4f = (Matrix4f)object;
-            return Float.compare(matrix4f.m00, this.m00) == 0 && Float.compare(matrix4f.m01, this.m01) == 0 && Float.compare(matrix4f.m02, this.m02) == 0 && Float.compare(matrix4f.m03, this.m03) == 0 && Float.compare(matrix4f.m10, this.m10) == 0 && Float.compare(matrix4f.m11, this.m11) == 0 && Float.compare(matrix4f.m12, this.m12) == 0 && Float.compare(matrix4f.m13, this.m13) == 0 && Float.compare(matrix4f.m20, this.m20) == 0 && Float.compare(matrix4f.m21, this.m21) == 0 && Float.compare(matrix4f.m22, this.m22) == 0 && Float.compare(matrix4f.m23, this.m23) == 0 && Float.compare(matrix4f.m30, this.m30) == 0 && Float.compare(matrix4f.m31, this.m31) == 0 && Float.compare(matrix4f.m32, this.m32) == 0 && Float.compare(matrix4f.m33, this.m33) == 0;
+            Matrix4f matrix4f = (Matrix4f) object;
+            return Float.compare(matrix4f.m00, this.m00) == 0
+                    && Float.compare(matrix4f.m01, this.m01) == 0
+                    && Float.compare(matrix4f.m02, this.m02) == 0
+                    && Float.compare(matrix4f.m03, this.m03) == 0
+                    && Float.compare(matrix4f.m10, this.m10) == 0
+                    && Float.compare(matrix4f.m11, this.m11) == 0
+                    && Float.compare(matrix4f.m12, this.m12) == 0
+                    && Float.compare(matrix4f.m13, this.m13) == 0
+                    && Float.compare(matrix4f.m20, this.m20) == 0
+                    && Float.compare(matrix4f.m21, this.m21) == 0
+                    && Float.compare(matrix4f.m22, this.m22) == 0
+                    && Float.compare(matrix4f.m23, this.m23) == 0
+                    && Float.compare(matrix4f.m30, this.m30) == 0
+                    && Float.compare(matrix4f.m31, this.m31) == 0
+                    && Float.compare(matrix4f.m32, this.m32) == 0
+                    && Float.compare(matrix4f.m33, this.m33) == 0;
```

Comment:
Yikes! :-o

Reply:
This PR has turned into me getting absolutely annoyed by rendering, even trying to port 1.12.2 rendering to 1.18 at some point. I will clean up this branch, force push, and once someone sets up a proper rendering framework, get back to it.

Reply:
Oh, rendering framework sounds like a challenge. :) I'll try to drop by discord when I have some time over, to understand what is needed to get proper rendering support for our needs in 1.18.

Reply:
I agree, if we could organize a meeting or such, even if it's text only, it would help a lot. 

Basically what we need is a ScreenRenderer like class from Wynntils (maybe improved somehow).  In the end, I believe that such system would help the development process greatly, as right now, I think this is the main holdback (at least for me personally)

Reply:
I had some time over this weekend that I was planning to spend on Artemis, I was "just" going to fix lingering bugs on my one-man mods (statuseffecttimer and ModSettings). That ended up taking all my spare time and then some. :-/

So I'm not sure when I can get time for Artimes development next time around. But I might be able to find some time for a brainstorming/discussion session on Discord.